src/main/resources/jnt_userRegistration/html/userRegistration.jsp

```diff
@@ -12,52 +12,17 @@
 <%--@elvariable id="renderContext" type="org.jahia.services.render.RenderContext"--%>
 <%--@elvariable id="currentResource" type="org.jahia.services.render.Resource"--%>
 <%--@elvariable id="url" type="org.jahia.services.render.URLGenerator"--%>
-<template:addResources type="javascript" resources="jquery.min.js,jquery.validate.js"/>
 <c:if test="${not renderContext.liveMode || not renderContext.loggedIn}">
     <script type="text/javascript">
-        $(document).ready(function() {
-            $("#newUser").validate({
-                rules: {
-                    desired_firstname : "required",
-                    desired_lastname: "required",
-                    username: {
-                        required: true,
-                        minlength: 2
-                    },
-                    password: {
-                        required: true,
-                        minlength: 6
-                    },
-                    confirm_password: {
-                        required: true,
-                        minlength: 6,
-                        equalTo: "#desired_password"
-                    },
-                    desired_email: {
-                        required: true,
-                        email: true
-                    }
-                },
-                messages: {
-                    desired_firstname: "<fmt:message key='userregistration.label.askFirstname'/>",
-                    desired_lastname: "<fmt:message key='userregistration.label.askLastname'/>",
-                    username: {
-                        required: "<fmt:message key='userregistration.label.askUsername'/>",
-                        minlength: "<fmt:message key='userregistration.label.usernameSizeWarning'/>"
-                    },
-                    password: {
-                        required: "<fmt:message key='userregistration.label.askPassword'/>",
-                        minlength: "<fmt:message key='userregistration.label.passwordSizeWarning'/>"
-                    },
-                    confirm_password: {
-                        required: "<fmt:message key='userregistration.label.askPassword'/>",
-                        minlength: "<fmt:message key='userregistration.label.passwordSizeWarning'/>",
-                        equalTo: "<fmt:message key='userregistration.label.validatePassword'/>"
-                    },
-                    desired_email: "<fmt:message key='userregistration.label.validateEmail'/>"
-                }
-            });
-        });
+        function onChange() {
```

Comment:
From [this](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity#examples) it says, we need to call `reportValidity` as well? 

Also doesn't seem to have support in IE <= 11 ([link](https://caniuse.com/mdn-api_htmlformelement_reportvalidity)). this is ok? (I'm a bit unclear what min browser support needed for these views).

Reply:
reportValidity() can be called to display the validity check on demand, but by default all forms does the validation when clicking on submit (unless you set the [novalidate](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#novalidate) attribute)

For IE<=11, no, this will not be supported, I don't know what is target for this code-sample/demo component . If we want to support old things, we may just do the check on server-side. (Note that I'm not even sure IE11 supports xhr)

Reply:
actually for the password recovery I'm not sure if we need xhr submit function at all. it looks to be just doing a validation on the username and then submitting the form.

we can probably add a validate and just use the default form submit. but we can also just keep as-is.

Reply:
xhr is used to avoid a redirection, and giving the result in a nice alert, so if we want to keep the behaviour we should keep it. Agree for the validate, the field should at least be required.